great_expectations/jupyter_ux/expectation_explorer.py

```diff
@@ -243,9 +244,9 @@ def initialize_data_asset_state(self, data_asset) -> None:
     def set_expectation_state(self, data_asset, expectation_state, column=None) -> None:
         data_asset_name = data_asset.data_asset_name
         expectation_type = expectation_state.get("expectation_type")
-        data_asset_state = self.state["data_assets"].get(data_asset_name)
+        data_asset_state: Dict = self.state["data_assets"][data_asset_name]
 
-        data_asset_expectations = data_asset_state.get("expectations")
+        data_asset_expectations = data_asset_state.get("expectations", {})
```

Comment:
These changes and the similar ones below assume that we aren't use an AttributeError exception for control flow.